cpp/test/sg/rf_treelite_test.cu

```diff
@@ -0,0 +1,403 @@
+struct RfInputs {
+  int n_rows;
+  int n_cols;
+  int n_trees;
+  float max_features;
+  float rows_sample;
+  int n_inference_rows;
+  int max_depth;
+  int max_leaves;
+  bool bootstrap;
+  bool bootstrap_features;
+  int n_bins;
+  int split_algo;
+  int min_rows_per_node;
+  CRITERION split_criterion;
+};
+
+template <typename T>
+::std::ostream& operator<<(::std::ostream& os, const RfInputs<T>& dims) {
+  return os;
+}
+
+template <typename T, typename L>
+class RfTreeliteTestCommon : public ::testing::TestWithParam<RfInputs<T>> {
+ protected:
+  void convertToTreelite() {
+    // Test the implementation for converting fitted forest into treelite format.
+    ModelHandle model;
+    build_treelite_forest(&model, forest, params.n_cols, task_category);
+
+    std::string test_name =
+      ::testing::UnitTest::GetInstance()->current_test_info()->name();
+    // Get the test index from Google current_test_info.
+    // The test index is the string after '/' in test_name.
+    std::string index_str =
+      test_name.substr(test_name.find("/") + 1, test_name.length());
+
+    // Create a directory if the test is the first one in the test case.
+    // https://stackoverflow.com/questions/12510874/how-can-i-check-if-a-directory-exists.
+    std::string create_dir = "mkdir " + test_dir;
+    struct stat info;
+    if (!(stat(test_dir.c_str(), &info) == 0 && S_ISDIR(info.st_mode))) {
+      LIBCCALL_CHECK(system(create_dir.c_str()));
+    }
+
+    // Create a sub-directory for the test case.
+    dir_name = test_dir + index_str;
+
+    CompilerHandle compiler;
+    // "ast_navive" is the default compiler treelite used in their Python code.
+    TREELITE_CHECK(TreeliteCompilerCreate("ast_native", &compiler));
+
+    int verbose = 0;
+    // Generate C code in the directory specified below.
+    TREELITE_CHECK(
+      TreeliteCompilerGenerateCode(compiler, model, verbose, dir_name.c_str()));
+    TREELITE_CHECK(TreeliteCompilerFree(compiler));
+
+    // Options copied from
+    // https://github.com/dmlc/treelite/blob/528d883f8f39eb5dd633e929b95915b63e210b39/python/treelite/contrib/__init__.py.
+    std::string obj_cmd = "gcc -c -O3 -o " + dir_name + "/main.o " + dir_name +
+                          "/main.c -fPIC "
+                          "-std=c99 -lm";
+
+    std::string lib_cmd = "gcc -shared -O3 -o " + dir_name +
+                          "/treelite_model.so " + dir_name +
+                          "/main.o -std=c99 -lm";
+
+    LIBCCALL_CHECK(system(obj_cmd.c_str()));
+    LIBCCALL_CHECK(system(lib_cmd.c_str()));
+
+    PredictorHandle predictor;
+    std::string lib_path = dir_name + "/treelite_model.so";
+
+    // -1 means use maximum possible worker threads.
+    int worker_thread = -1;
+    TREELITE_CHECK(
+      TreelitePredictorLoad(lib_path.c_str(), worker_thread, &predictor));
+
+    DenseBatchHandle dense_batch;
+    // Current RF dosen't seem to support missing value, put FLT_MAX to be safe.
+    float missing_value = FLT_MAX;
```

Comment:
Typically, NaN is used as a missing value. However, feel free to use any missing value as long as treelite supports it.